plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -250,6 +272,46 @@ public Schema getTableSchema(String table)
         return sendHttpGetToControllerJson(format(TABLE_SCHEMA_API_TEMPLATE, table), schemaJsonCodec);
     }
 
+    public List<String> getPinotTableNames()
+    {
+        return ImmutableList.copyOf(getFromCache(allTablesCache, ALL_TABLES_CACHE_KEY).keySet());
+    }
+
+    public static <K, V> V getFromCache(LoadingCache<K, V> cache, K key)
+    {
+        V value = cache.getIfPresent(key);
+        if (value != null) {
+            return value;
+        }
+        try {
+            return cache.get(key);
+        }
```

Comment:
why do this manually? Won't a `cache.get` return from cache if loaded otherwise execute the loader? Do I miss something.

Reply:
iirc, this is from the initial version of the pinot connector, I think it was intended as an optimization (but I don't see the benefit) but to be sure would require some testing, maybe it minimized races when fetching (just guessing tbh). Should it I try that and test it for another pr or do you think we should just try it here and see if it causes any issues? Either way works for me, lmk.

Reply:
Let's keep it for now since it was pre-existing.